File: android/app/src/main/java/com/zindolla/radioamoris/Station.java
package com.zindolla.myradio;

import android.os.Parcel;
import android.os.Parcelable;

public class Station implements Parcelable {
    int id;
    String name;
    String logo;
    String url;

    private Station(Parcel in) {
        id = in.readInt();
        name = in.readString();
        logo = in.readString();
        url = in.readString();
    }

    Station(int id, String name, String logo, String url) {
        this.id = id;
        this.name = name;
        this.logo = logo;
        this.url = url;
    }

    public static final Creator<Station> CREATOR = new Creator<Station>() {
        @Override
        public Station createFromParcel(Parcel in) {
            return new Station(in);
        }

        @Override
        public Station[] newArray(int size) {
            return new Station[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeInt(id);
        dest.writeString(name);
        dest.writeString(logo);
        dest.writeString(url);
    }
}

File: android/app/src/main/kotlin/com/zindolla/radioamoris/Station.kt
package com.zindolla.radioamoris

import android.os.Parcel
import android.os.Parcelable

class Station : Parcelable {
    var id: Int
    var name: String?
    var logo: String?
    var url: String?

    private constructor(`in`: Parcel) {
        id = `in`.readInt()
        name = `in`.readString()
        logo = `in`.readString()
        url = `in`.readString()
    }

    constructor(id: Int, name: String?, logo: String?, url: String?) {
        this.id = id
        this.name = name
        this.logo = logo
        this.url = url
    }

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeInt(id)
        dest.writeString(name)
        dest.writeString(logo)
        dest.writeString(url)
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<Station> = object : Parcelable.Creator<Station> {
            override fun createFromParcel(`in`: Parcel): Station {
                return Station(`in`)
            }

            override fun newArray(size: Int): Array<Station?> {
                return arrayOfNulls(size)
            }
        }
    }
}
